python/ray/data/read_api.py

```diff
@@ -186,6 +196,7 @@ def read_datasource(
     *,
     parallelism: int = 200,
     ray_remote_args: Dict[str, Any] = None,
+    _dataset_name: str = "",
```

Comment:
As a follow-up (definitely not in this PR), might be worth delegating the default dataset name to the datasource. E.g., maybe a `FileBasedDatasource` would use the base name of the provided directory as the default dataset name.